### ISSUE TITLE
Grafiek: aantal geïntroduceerde uitheemse soorten per jaar per regio van oorsprong

### ISSUE THREAD

**Issue by SanderDevisscher** [inbo/reporting-rshiny-grofwildjacht#145]
Gebaseerd op deze indicator: https://trias-project.github.io/indicators/indicator_introductions_per_year.html
maar in de stijl van `FIGUUR: Gerapporteerd aantal per jaar en per regio` van de grofwildjacht pagina. 
Dus per jaar een stacked bar chart met de verschillende regios

Met deze data: https://github.com/trias-project/indicators/blob/master/data/interim/data_input_checklist_indicators.tsv

en de volgende selectie opties:
- [ ]  periode (slider)
- [ ]  taxonomy (kingdom, phylum, ... t.e.m. family)
- [ ]  locality (België (BE), Vlaanderen (BE-VLG), Brussel (BE-BRU), Wallonië (BE-WAL))
- [ ]  degree of establishment
- [ ]  pathway_level 1 & pathway_level 2
- [ ]  habitat
- [ ]  source

**Comment by mvarewyck** [comment]
@timadriaens @damianooldoni "per regio van oorsprong", is this variable "native range" in the data?
Because I found 150 different values for that - which might not be ideal to show in stacked barplot
```
> unique(exotenData[, "native range"])
                                            native range
  1:                                                    
  2:                                       North America
  3:                                 North Pacific Ocean
  4:                            Eastern Asia (WGSRPD:38)
  5:                                   Europe (WGSRPD:1)
 ---                                                    
146:                                             Eurasia
147:                                       central Italy
148:                                       western Italy
149: northern coastal areas of the western Mediterranean
150:                            Mediterranean & Portugal
```
Also, multiple values are separated by "," or "&", not by "|" as indicated https://docs.google.com/spreadsheets/d/1LeXXbry2ArK2rngsmFjz_xErwE1KwQ8ujtvHNmTVA6E/edit#gid=808615322

**Comment by SanderDevisscher** [comment]
@damianooldoni @mvarewyck 

How hard is it to translate native range into this ? analog to habitat being translated into marine, freshwater, terrestrial.

| native range                                        | Azië  | Afika | Noord-Amerika | Zuid-Amerika | Antartica | Europa | Oceanië |
|-----------------------------------------------------|-------|-------|---------------|--------------|-----------|--------|---------|
| North America                                       | FALSE | FALSE | TRUE          | FALSE        | FALSE     | FALSE  | FALSE   |
| North Pacific Ocean                                 | FALSE | FALSE | FALSE         | FALSE        | FALSE     | FALSE  | TRUE    |
| Eastern Asia (WGSRPD                                | FALSE | FALSE | FALSE         | FALSE        | FALSE     | FALSE  | FALSE   |
| Europe (WGSRPD                                      | FALSE | FALSE | FALSE         | FALSE        | FALSE     | TRUE   | FALSE   |
| Eurasia                                             | TRUE  | FALSE | FALSE         | FALSE        | FALSE     | TRUE   | FALSE   |
| central Italy                                       | FALSE | FALSE | FALSE         | FALSE        | FALSE     | TRUE   | FALSE   |
| western Italy                                       | FALSE | FALSE | FALSE         | FALSE        | FALSE     | TRUE   | FALSE   |
| northern coastal areas of the western Mediterranean | FALSE | FALSE | FALSE         | FALSE        | FALSE     | TRUE   | FALSE   |
| Mediterranean & Portugal                            | FALSE | FALSE | FALSE         | FALSE        | FALSE     | TRUE   | FALSE   |

**Comment by SanderDevisscher** [comment]
> Also, multiple values are separated by "," or "&", not by "|" as indicated https://docs.google.com/spreadsheets/d/1LeXXbry2ArK2rngsmFjz_xErwE1KwQ8ujtvHNmTVA6E/edit#gid=808615322


@damianooldoni can you look into this ?

**Comment by timadriaens** [comment]
yes, that's one of the problems with the current checklist, in principle they should have been mapped at checklist level using the World Geographical Scheme for Recording Plant Species Distribution (WGSRPD, https://web.archive.org/web/20160125135239/http://www.nhm.ac.uk/hosted_sites/tdwg/TDWG_geo2.pdf) and the UN geoscheme (https://en.wikipedia.org/wiki/United_Nations_geoscheme) for other species.

**Comment by damianooldoni** [comment]
@SanderDevisscher: it's a long recoding and it's just tedious to write it:

```
df %>%
  mutate(Azië = if_else(native_range == "Eurasia", TRUE, FALSE),
              Afrika = FALSE, #no TRUE values in table above
              Noord-Amerika = if_else(native_range == "North America", TRUE, FALSE),
              etc.
)
```

@timadriaens, should we do this mapping for TrIAS indicators report as well? Or it's something you want only for RShyny visualization webpage?

**Comment by damianooldoni** [comment]
@timadriaens : I ask again our TrIAS colleagues about possibility of mapping it at checklist publication level or while unifiying the checklists.

**Comment by timadriaens** [comment]
@damianooldoni I think very often, checklists will have more detailed (and not very standardized) information on native range. And it would be "a pitty" (and not very stimulating to the checklist owner to keep track of such detail) actually if this went lost at checklist publication level. I don't know what is the best place - and there is always the option of keeping a verbatim field - but it is us who have made the very pragmatic decision to use WGSRPD for plants and UN geoscheme for animals, these are very crude anyway.

**Comment by timadriaens** [comment]
So in the logic of preserving as much detail in the source checklists as possible, I would say it is more something for the flow to the unified. And of course, this should first of all be in the TrIAS indicators.

**Comment by SanderDevisscher** [comment]
> @SanderDevisscher: it's a long recoding and it's just tedious to write it:
> 
> ```
> df %>%
>   mutate(Azië = if_else(native_range == "Eurasia", TRUE, FALSE),
>          Afrika = FALSE, #no TRUE values in table above
>          Noord-Amerika = if_else(native_range == "North America", TRUE, FALSE),
>          etc.
> )
> ```
> 
> @timadriaens, should we do this mapping for TrIAS indicators report as well? Or it's something you want only for RShyny visualization webpage?

@damianooldoni what about grepl ? or a manual mapping list ?

**Comment by damianooldoni** [comment]
@SanderDevisscher : yes, a file containing the mapping seems way better.

**Comment by peterdesmet** [comment]
Is the UN Geocode scheme this one: https://en.wikipedia.org/wiki/United_Nations_geoscheme?

If we're going to standardize it anyway, why not use one scheme for all?

**Comment by timadriaens** [comment]
because the botanists wanted their own thing see [this issue](https://github.com/trias-project/ad-hoc-checklist/issues/10)

**Comment by timadriaens** [comment]
there is of course also the issue of marine species where native range requires a dedicated vocab

**Comment by peterdesmet** [comment]
Regarding botanists: well, I think we (authors of the unified checklist) can choose to which extend we want to aggregate (including using the same scheme for plants and animals, marine taxa are indeed separate). The source data will always be available. So, it comes down to making a decision.

**Comment by eadriaensen** [comment]
@SanderDevisscher @timadriaens @damianooldoni 
I assume dealing with `native_range` is still ongoing?

Could you update whenever this is ready for me to continue working on the plot?

**Comment by SanderDevisscher** [comment]
@damianooldoni what is the status of the `native range` issue ?

**Comment by timadriaens** [comment]
values have largely been standardized by @peterdesmet see [this issue](https://github.com/trias-project/unified-checklist/issues/11#issuecomment-608453484) but not sure this is already updated in the unified @damianooldoni ?

**Comment by SanderDevisscher** [comment]
@damianooldoni can you make sure the mapping find its way into data_input_checklist_indicators.tsv

**Comment by peterdesmet** [comment]
It has not been included in the unified checklist, because:

- The mapping is not yet complete
- It has not been decided if the mapping should be part of the indicators or unified checklist @timadriaens?

**Comment by timadriaens** [comment]
As said and argumented [here](https://github.com/trias-project/unified-checklist/issues/11#issuecomment-608453484), I definitely think they should be part of it, yes. Otherwise, breakdowns of indicators based on native ranges are impossible.

**Comment by peterdesmet** [comment]
This is now republished with standardized native range in the unified checklist. Can you reprocess and see if that solve the issue?

**Comment by damianooldoni** [comment]
Thanks @peterdesmet . I will asap process unified checklist data to include these changes.

**Comment by eadriaensen** [comment]
@SanderDevisscher @timadriaens @damianooldoni 
I imported the update of the checklist. So now there are 25 different values for `native_range`.

```
> table(exotenData[, "native_range"], exclude = NULL)

                   Africa                  Americas                      Asia 
                     1380                       211                       566 
Australia and New Zealand                 Caribbean           Central America 
                      192                         5                        25 
             Central Asia            Eastern Africa              Eastern Asia 
                       13                        24                      2549 
           Eastern Europe                    Europe                 Melanesia 
                     2587                      2511                         1 
               Micronesia             Middle Africa           Northern Africa 
                        2                        14                        23 
         Northern America             South America         Southeastern Asia 
                     1205                       553                        52 
          Southern Africa             Southern Asia           Southern Europe 
                       15                        75                        83 
           Western Africa              Western Asia            Western Europe 
                       19                        28                        18 
                     <NA> 
                      603 
```
Concerning the requested graph:
> maar in de stijl van FIGUUR: Gerapporteerd aantal per jaar en per regio van de grofwildjacht pagina.
Dus per jaar een stacked bar chart met de verschillende regios

Do I write the code for this graph?
If so, be aware that the style might differ from the graphs that are currently used for _aantal geïntroduceerde uitheemse soorten per jaar_ and _Cumulatief aantal uitheemse soorten_ as they come from the TRiAS package.
Or is there code available?

**Comment by SanderDevisscher** [comment]
> Do I write the code for this graph?

Yes, if it were up to me!

> If so, be aware that the style might differ from the graphs that are currently used for aantal geïntroduceerde uitheemse soorten per jaar and Cumulatief aantal uitheemse soorten as they come from the TRiAS package.

I think it is best we overwrite the TrIAS style with the Inbotheme style or at least use a similar style as the rest of the app. 

> not that I know of. Shall I make a basis for you to start from or would you like to start from scratch ?

**Comment by peterdesmet** [comment]
Note: the regions can be grouped in 5 continents: https://en.wikipedia.org/wiki/United_Nations_geoscheme

**Comment by eadriaensen** [comment]
> If so, be aware that the style might differ from the graphs that are currently used for aantal geïntroduceerde uitheemse soorten per jaar and Cumulatief aantal uitheemse soorten as they come from the TRiAS package.
>>I think it is best we overwrite the TrIAS style with the Inbotheme style or at least use a similar style as the rest of the app.

Okay, I will give that a go.

>Or is there code available?
>>Not that I know of. Shall I make a basis for you to start from or would you like to start from scratch ?
Thanks, I have the code from grofwildjacht to start from. So it's okay

**Comment by eadriaensen** [comment]
> Note: the regions can be grouped in 5 continents: https://en.wikipedia.org/wiki/United_Nations_geoscheme


Do you prefer to display on continent level (in that case I regroup)? Or on `native_range` level as in raw data?

**Comment by SanderDevisscher** [comment]
> > Note: the regions can be grouped in 5 continents: https://en.wikipedia.org/wiki/United_Nations_geoscheme
> 
> Do you prefer to display on continent level (in that case I regroup)? Or on `native_range` level as in raw data?

Maybe add a unit type filter or a switch (to increase details-level)

**Comment by eadriaensen** [comment]
First version of the plot:

![exotenPerYearPerNative_continent](https://user-images.githubusercontent.com/40824010/79565087-3a5d1680-80b0-11ea-9b8e-bc7833d64c18.png)

**Comment by SanderDevisscher** [comment]
> First version of the plot:
> 
> ![exotenPerYearPerNative_continent](https://user-images.githubusercontent.com/40824010/79565087-3a5d1680-80b0-11ea-9b8e-bc7833d64c18.png)

Ziet er goed uit maar wel lage aantallen, Is dit gefilterd ?

**Comment by SanderDevisscher** [comment]
@mvarewyck @eadriaensen is it possible to add an option to choose between the relative and absolute numbers for this graph ?

**Comment by eadriaensen** [comment]
It would be good to implement the code for this plot in the trias package, for sake of uniformity - and given the limited remaining project hours.
In that case, the option to display relative or absolute numbers should ideally be implemented in the function there as an argument.

If so, I can easily implement a tick box for the user in the app to switch between relative and absolute

The current code, used for the graph :
[exoten_indicatorYearByNativeRange.zip](https://github.com/inbo/reporting-rshiny-grofwildjacht/files/4851491/exoten_indicatorYearByNativeRange.zip)

@SanderDevisscher is this possible?

**Comment by SanderDevisscher** [comment]
@damianooldoni can you look into this ?

**Comment by damianooldoni** [comment]
I agree with @eadriaensen; it is better to develop this functionality in the `trias` package _for sake of uniformity_. 

But I go on holiday on Thursday. No time to do it this afternoon or tomorrow.
As maintainer of the `trias` package I am more than happy to accept contributions and PRs. @SanderDevisscher or somebody else involved: maybe have you some time to give a try? I can review it when I come back (~2 weeks). And in the meanwhile you can use the trias package in development branch for the shiny app instead of the master?
Thanks

**Comment by SanderDevisscher** [comment]
@damianooldoni I'll give it a try
